### DISCUSSION TITLE
Https authorization to github

### DISCUSSION THREAD

**Discussion by SebastianMpl** [jj-vcs/jj#2724] [Q&A] [answered]
Hi,
Few days ago I downloaded the jj binary and currently trying to make it working with Github.
I am having some problems with ssh but I'll deal with them later.
Currently I want to ask:
Cloning repository via http and pushing it back to github jj asks me to type username and passphrase.
I assume that Username is my github username - not email and the passphrase is the personal access token generated in my github profile settings?
But writing them by hand every time is boring. Where and how should I save them so they will be taken by jj automatically?

**Comment by martinvonz** [comment, marked as answer]
I never use http authentication myself, but I think it should be the same username and password as you would use with `git push`.

I think most people use SSH for pushing, and usually combined with `ssh-agent`. The instructions for that seem to be here: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent.

It looks like we don't have a single command for updating the remote URL associated with a remote, so I think you'll have to do `jj git remote remove origin` followed by `jj git remote add origin <SSH URL copied from GitHub repo>`.

**Comment by SebastianMpl** [comment]
@martinvonz

Thank you for your answer. Meanwhile I managed to configure ssh keys and indeed everything works fine.
But I need to remember the commands in the last line of your answer.

Thank you,
